### PR TITLE
Update links in initializers documentation

### PR DESCRIPTION
**Pull request by himkt**
I found some links in the initializer documentation were broken.
This PR updates them to fix.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] ~I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.~
- [x] ~If this PR fixes a bug, I've added a test that will fail without my fix.~
- [x] ~If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.~

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.
- [ ] **`codecov/patch`** reports high test coverage (at least 90%).
    You can find this under the "Actions" tab of the pull request once the other checks have finished.

### CODE REVIEW

File: allennlp/nn/initializers.py [hunk 1]
```diff
@@ -7,20 +7,20 @@
...
-* ["kaiming_uniform"](https://pytorch.org/docs/master/nn.html?highlight=orthogonal#torch.nn.init.kaiming_uniform_)
-* ["kaiming_normal"](https://pytorch.org/docs/master/nn.html?highlight=orthogonal#torch.nn.init.kaiming_normal_)
-* ["orthogonal"](https://pytorch.org/docs/master/nn.html?highlight=orthogonal#torch.nn.init.orthogonal_)
-* ["sparse"](https://pytorch.org/docs/master/nn.html?highlight=orthogonal#torch.nn.init.sparse_)
+* ["normal"](https://pytorch.org/docs/master/nn.init.html?highlight=uniform#torch.nn.init.normal_)
+* ["uniform"](https://pytorch.org/docs/master/nn.init.html?highlight=orthogonal#torch.nn.init.uniform_)
+* ["constant"](https://pytorch.org/docs/master/nn.init.html?highlight=orthogonal#torch.nn.init.constant_)
+* ["eye"](https://pytorch.org/docs/master/nn.init.html?highlight=orthogonal#torch.nn.init.eye_)
+* ["dirac"](https://pytorch.org/docs/master/nn.init.html?highlight=orthogonal#torch.nn.init.dirac_)
+* ["xavier_uniform"](https://pytorch.org/docs/master/nn.init.html?highlight=orthogonal#torch.nn.init.xavier_uniform_)
```
Comment (himkt) [inline allennlp/nn/initializers.py:15]: The length of the line exceeds 115... 💭

Comment (epwalsh) [reply allennlp/nn/initializers.py:15]: You could just add a `# noqa: E501` comment to the end of this docstring. See https://github.com/allenai/allennlp/blob/56e1f49df918cff52707d64dcf894885250aaf89/allennlp/modules/token_embedders/pretrained_transformer_embedder.py#L73, for example.

**Review by epwalsh — COMMENTED** [review]
LGTM other than the linting issue!

File: allennlp/nn/initializers.py (hunk 1 above)
Comment (himkt) [reply allennlp/nn/initializers.py:15]: It makes sense! I added the comment in e565dcf.

**Review by epwalsh — APPROVED** [review]
Thanks @himkt!

**Pull request merged**